### PR TITLE
Feature/event auth

### PR DESCRIPTION
**Pull request by WhoaNellie**
Events are using authentication, so events are linked to specific, signed in, users!

Based off of my user controllers branch so I should wait to merge this until that one's merged.

### CODE REVIEW

**Review by alvyjudy — APPROVED** [review]
Looks good to me!

File: server/routes/eventAPI.js [hunk 1]
```diff
@@ -1,33 +1,31 @@
...
-        res.status(500).send(error);
-      });
-  } else if (!req.body.user_id) {
-    res.status(400).send("User ID is required");
-  } else if (!req.body.duration) {
-    res.status(400).send("Duration is required");
-  }
+router.post("/api/event", auth, (req, res) => {
+    if (req.body.duration) {
+        db.EventType.create({ ...req.body, user_id: req.userId })
```
Comment (dproc96) [inline server/routes/eventAPI.js:10]: user_id should be camelcase not snakecase, looking into the commits I think this is from a PR that Sina and I didn't review

File: server/routes/userAPI.js [hunk 2]
```diff
@@ -6,8 +6,7 @@ const mongoose = require("mongoose");
 const router = express.Router();
 
 // GET whether id is unique
-router.get("/api/user/is_unique", (req, res) => {
-
+router.get("/api/user/is_unique", auth, (req, res) => {
```
Comment (dproc96) [inline server/routes/userAPI.js:9]: Dashes are [preferred](https://developers.google.com/search/docs/advanced/guidelines/url-structure?hl=en&visit_id=637467458802697471-4015835834&rd=1) to underscores in urls

**Review by dproc96 — APPROVED** [review]
I noticed some issues with conventions, but they are unrelated to this PR so we should probably handle those in a separate PR. This one is good to go!

**Pull request merged**